transport/grpc/outbound_test.go

```diff
@@ -200,3 +203,159 @@ func TestCallStreamWithInvalidPeer(t *testing.T) {
 		}.Error(),
 	)
 }
+
+func TestCallSvcMatchSuccess(t *testing.T) {
+	matchServer := grpc.NewServer(
+		grpc.CustomCodec(customCodec{}),
+		grpc.UnknownServiceHandler(
+			func(srv interface{}, stream grpc.ServerStream) error {
+				ctx := stream.Context()
+				// Get transport request
+				md, ok := metadata.FromIncomingContext(ctx)
+				if md == nil || !ok {
+					return yarpcerrors.Newf(yarpcerrors.CodeInternal, "cannot get metadata from ctx: %v", ctx)
+				}
+				transportRequest, err := metadataToTransportRequest(md)
+				if err != nil {
+					return err
+				}
+
+				responseWriter := newResponseWriter()
+				defer responseWriter.Close()
+
+				// Echo accepted rpc-service in response header
+				responseWriter.AddSystemHeader(ServiceHeader, transportRequest.Service)
+
+				// Send the response attributes back and end the stream.
+				if sendErr := stream.SendMsg(responseWriter.Bytes()); sendErr != nil {
+					// We couldn't send the response.
+					return sendErr
+				}
+				if responseWriter.md != nil {
+					stream.SetTrailer(responseWriter.md)
+				}
+				return err
+			}),
+	)
+	listener, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:15213"))
+	require.NoError(t, err)
+	go func() {
+		// TODO: do we have consistency issue here? Request comes before server begin to serve
+		_ = matchServer.Serve(listener)
```

Comment:
let's add a `require.NoError(err)` on the returned error instead of ignoring it

Reply:
Added, the source code in [transport/grpc/inbound.go](https://sourcegraph.com/github.com/yarpc/yarpc-go/-/blob/transport/grpc/inbound.go#L134:31) maybe also need modify.